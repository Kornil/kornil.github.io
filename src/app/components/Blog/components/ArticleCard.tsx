import React from "react";
import styled from "react-emotion";

import { StyledEm } from "app/styled-components";
import { getDateFromUnix } from "app/utils";

import StoryInterface from "../interfaces/StoryInterface";

const StyledArticleCard = styled.section`
  margin-bottom: 48px;

  a {
    color: #3d3d3d;
    text-decoration: none;

    :hover {
      color: #1f1f1f;
    }
  }

  h3 {
    font-weight: normal;
    margin: 8px 0 12px;
  }

  p {
    margin: 0;
    font-size: 14px;
  }
`;

const ArticleCard: React.SFC<{ story: StoryInterface }> = ({ story }) => (
  <StyledArticleCard>
    <a href={story.link}>
      <h3>{story.title}</h3>
      <p>
        <StyledEm>{getDateFromUnix(story.publishedAt)}</StyledEm>
      </p>
    </a>
  </StyledArticleCard>
);

export default ArticleCard;
