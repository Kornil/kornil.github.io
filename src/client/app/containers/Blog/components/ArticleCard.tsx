import React from "react";
import styled from "react-emotion";

import { StyledLink } from "app/styled-components";
import { getDateFromUnix } from "app/utils";

import { StoryInterface } from "app/context/StoriesContext";

const StyledArticleCard = styled.section`
  margin-bottom: 48px;

  h3 {
    font-weight: normal;
    margin: 8px 0 12px;
  }

  p {
    margin: 0;
    font-size: 14px;
  }
`;

const ArticleCard: React.SFC<{ story: StoryInterface }> = ({ story }) => (
  <StyledArticleCard>
    <StyledLink
      to={`/blog/${story.link.replace(
        "https://medium.com/@francesco.agnoletto/",
        ""
      )}`}
    >
      <h3>{story.title}</h3>
      <p>
        <em>{getDateFromUnix(story.publishedAt)}</em>
      </p>
    </StyledLink>
  </StyledArticleCard>
);

export default ArticleCard;
