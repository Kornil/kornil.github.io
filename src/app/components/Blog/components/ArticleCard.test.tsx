import { shallow } from "enzyme";
import React from "react";

import ArticleCard from "./ArticleCard";

const story = {
  link: "www.google.com",
  publishedAt: 1530736049131,
  title: "hello"
};

it("it renders", () => {
  const wrapper = shallow(<ArticleCard story={story} />);
  expect(wrapper).toMatchSnapshot();
});
