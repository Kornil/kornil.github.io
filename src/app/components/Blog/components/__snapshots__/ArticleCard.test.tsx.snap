// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<StyledArticleCard>
  <a
    href="www.google.com"
  >
    <h3>
      hello
    </h3>
    <p>
      <StyledEm>
        04/07/2018
      </StyledEm>
    </p>
  </a>
</StyledArticleCard>
`;
